### ISSUE TITLE
punch_slug breaks really_destroy!

### ISSUE THREAD

**Issue by brianmcmahon** [spree/spree#6389]
Hi,
I'm running rails 4.1.8 with spree 2.4.2 and it seems like the punch slug method breaks using really_destroy on the products. This seems to makes sense since it's trying to update the record after you delete it, so the error "ActiveRecord::ActiveRecordError: cannot update on a new record object" makes sense. I'm still playing around with how I want it fixed, but a suggestion might be checking if it's a new records before running the update.

**Comment by JDutil** [comment]
I would try updating to the latest 2-4-stable as I think punch slug changes have occurred, but not sure that they will actually help your use case.  Closing due to a lack of information, and you using outdated rails and 2-4-stable that may or may not help you solve your problem. close:steps

**Comment by spreebot** [comment]
You need to provide a detailed set of steps to reproduce the problem. It should be possible for another developer to reproduce the problem using a clean install (no extensions or customizations) if they follow your steps and use the Spree version specified.

Don't get discouraged that your issue has been closed, Spree is a busy project and gets a lot of issues so we need your help to make sure this issue contains all the information required. Please review [our guidelines](http://dev.spreecommerce.com/issues#reporting-requirements) on how to submit an issue.

**Comment by brianmcmahon** [comment]
Hey Jeff,
I updated to 2-4-stable and the issue is still there, though it's a different error message. So this is reproducible by creating a product then calling really_destroy! on it in the rails console. You get an error message "can't destroy frozen hash". The issue seems to lie within 'punch_slug' in the Spree::Product model. The issue looks fixed for 3-0-stable, since you added the frozen? check to the method. I added that check to the method and it fixed the issue.
